includes/UserMailer.php

```diff
@@ -550,203 +603,148 @@ public function notifyOnPageChange( $editor, $title, $timestamp, $summary, $mino
 	 * Also updates wl_notificationtimestamp.
 	 *
-	 * @param $editor User object
-	 * @param $title Title object
-	 * @param $timestamp string Edit timestamp
-	 * @param $summary string Edit summary
-	 * @param $minorEdit bool
-	 * @param $oldid int Revision ID
-	 * @param $watchers array of user IDs
-	 * @param $action (Wikia)
-	 * @param $otherParam
 	 */
-	public function actuallyNotifyOnPageChange( $editor, $title, $timestamp, $summary, $minorEdit, $oldid, $watchers, $action='', $otherParam = array() ) {
-		# we use $wgPasswordSender as sender's address
+	private function actuallyNotifyOnPageChange( $watchers ) {
 		global $wgEnotifWatchlist;
-		global $wgEnotifMinorEdits, $wgEnotifUserTalk;
 
-		wfProfileIn( __METHOD__ );
+		$this->setReplyToAndFromAddresses();
 
 		# The following code is only run, if several conditions are met:
 		# 1. EmailNotification for pages (other than user_talk pages) must be enabled
 		# 2. minor edits (changes) are only regarded if the global flag indicates so
+		if ( !$this->isMinorEdit() || ( $this->notifyUsersOnMinorEdits() && $this->editorWantsToNotifyOnMinorEdits() ) ) {
 
-		$isUserTalkPage = ( $title->getNamespace() == NS_USER_TALK );
-
-		$this->title = $title;
-		$this->timestamp = $timestamp;
-		$this->summary = $summary;
-		$this->minorEdit = $minorEdit;
-		$this->oldid = $oldid;
-		$this->action = $action;
-		$this->editor = $editor;
-		$this->composed_common = false;
-		$this->other_param = $otherParam;
-		
-		$userTalkId = false;
-
-		if ( !$minorEdit || ( $wgEnotifMinorEdits && !$editor->isAllowed( 'nominornewtalk' ) ) ) {
-
-			if ( $wgEnotifUserTalk && $isUserTalkPage && $this->canSendUserTalkEmail( $editor, $title, $minorEdit ) ) {
-				$targetUser = User::newFromName( $title->getText() );
+			$userTalkId = 0;
+			if ( $this->isUserTalkPage() && $this->canSendUserTalkEmail() ) {
+				$targetUser = User::newFromName( $this->title->getText() );
 				$this->compose( $targetUser );
 				$userTalkId = $targetUser->getId();
 
-				/* Wikia change begin - @author: Marooned */
-				/* Send mail to user when comment on his user talk has been added - see RT#44830 */
+				// Send mail to user when comment on his user talk has been added
 				$fakeUser = null;
-				wfRunHooks('UserMailer::NotifyUser', array( $title, &$fakeUser ));
+				wfRunHooks( 'UserMailer::NotifyUser', [ $this->title, &$fakeUser ] );
 				if ( $fakeUser instanceof User && $fakeUser->getOption( 'enotifusertalkpages' ) && $fakeUser->isEmailConfirmed() ) {
-					wfDebug( __METHOD__.": sending talk page update notification\n" );
 					$this->compose( $fakeUser );
 				}
-				/* Wikia change end */
 			}
 
 			if ( $wgEnotifWatchlist ) {
 				// Send updates to watchers other than the current editor
 				$userArray = UserArray::newFromIDs( $watchers );
 
+				/* @var $watchingUser User */
 				foreach ( $userArray as $watchingUser ) {
 					if ( $watchingUser->getOption( 'enotifwatchlistpages' ) &&
```

Comment:
Another round about ```if```.  These guys sure loved length conditionals!  This would be another good candidate for its own function, with each conditional spelled out as above.  However could happen when this logic is moved to the Email extension.  Maybe for now just moving the ```&&``` on line 641 up to the end of the previous line for consistency sake is enough.